src/ray/gcs/gcs_server/test/gcs_actor_manager_test.cc

```diff
@@ -190,6 +193,64 @@ TEST_F(GcsActorManagerTest, TestActorReconstruction) {
   ASSERT_EQ(mock_actor_scheduler_->actors.size(), 0);
 }
 
+TEST_F(GcsActorManagerTest, TestNamedActors) {
+  auto job_id_1 = JobID::FromInt(1);
+  auto job_id_2 = JobID::FromInt(2);
+
+  auto request1 =
+      Mocker::GenCreateActorRequest(job_id_1, 0, /*is_detached=*/true, /*name=*/"actor1");
+  Status status = gcs_actor_manager_.RegisterActor(
+      request1, [](std::shared_ptr<gcs::GcsActor> actor) {});
+  ASSERT_TRUE(status.ok());
+  ASSERT_EQ(gcs_actor_manager_.GetNamedActorID("actor1").Binary(),
+            request1.task_spec().actor_creation_task_spec().actor_id());
+
+  auto request2 =
+      Mocker::GenCreateActorRequest(job_id_1, 0, /*is_detached=*/true, /*name=*/"actor2");
+  status = gcs_actor_manager_.RegisterActor(request2,
+                                            [](std::shared_ptr<gcs::GcsActor> actor) {});
+  ASSERT_TRUE(status.ok());
+  ASSERT_EQ(gcs_actor_manager_.GetNamedActorID("actor2").Binary(),
+            request2.task_spec().actor_creation_task_spec().actor_id());
+
+  // Check that looking up a non-existent name returns ActorID::Nil();
+  ASSERT_EQ(gcs_actor_manager_.GetNamedActorID("actor3"), ActorID::Nil());
+
+  // Check that naming collisions return Status::Invalid.
+  auto request3 =
+      Mocker::GenCreateActorRequest(job_id_1, 0, /*is_detached=*/true, /*name=*/"actor2");
+  status = gcs_actor_manager_.RegisterActor(request3,
+                                            [](std::shared_ptr<gcs::GcsActor> actor) {});
+  ASSERT_TRUE(status.IsInvalid());
+  ASSERT_EQ(gcs_actor_manager_.GetNamedActorID("actor2").Binary(),
+            request2.task_spec().actor_creation_task_spec().actor_id());
+
+  // Check that naming collisions are enforced across JobIDs.
+  auto request4 =
+      Mocker::GenCreateActorRequest(job_id_2, 0, /*is_detached=*/true, /*name=*/"actor2");
+  status = gcs_actor_manager_.RegisterActor(request4,
+                                            [](std::shared_ptr<gcs::GcsActor> actor) {});
+  ASSERT_TRUE(status.IsInvalid());
+  ASSERT_EQ(gcs_actor_manager_.GetNamedActorID("actor2").Binary(),
+            request2.task_spec().actor_creation_task_spec().actor_id());
+
+  // Check that naming only applies to detached actors.
+  auto request5 = Mocker::GenCreateActorRequest(job_id_1, 0, /*is_detached=*/false,
+                                                /*name=*/"actor2");
```

Comment:
It seems a bit confusing that you can pass in a name for a non-detached actor, but that it won't do anything. Is that just to support the existing API?

Reply:
I'll change this to a check.